vendor/github.com/waku-org/go-waku/waku/v2/protocol/subscription/subscription_details.go

```diff
@@ -107,6 +108,15 @@ func (s *SubscriptionDetails) Close() error {
 	return s.mapRef.Delete(s)
 }
 
+func (s *SubscriptionDetails) SetClosing() {
+	s.Lock()
+	defer s.Unlock()
+	if !s.Closed {
+		s.Closed = true
```

Comment:
I think this is ok - `Closed` will prevent this (https://github.com/waku-org/go-waku/blob/master/waku/v2/protocol/subscription/subscriptions_map.go#L202) from writing another envelope into a closing subscribtion and then it will clean up stuff by writing to the channel